src/components/input-field/input-field.tsx

```diff
@@ -570,4 +574,17 @@ export class InputField {
             event.preventDefault();
         }
     }
+
+    private syncInvalidToTextField() {
+        if (!this.mdcTextField) {
+            return;
+        }
+
+        if (!this.mdcTextField.valid) {
+            return;
+        }
+
+        const isValid = !this.isInvalid();
```

Comment:
The `isInvalid` function looks a little weird to me, especially when `syncInvalidToTextField` is added to the mix.

Seems like we're now juggling the `invalid` property set by the consumer, the validity reported by the *native* element, and the validity reported by the MDC component, and it's spread out over two different functions where one is calling the other, but both are called by other code at different points in the life cycle. I think it's too complex and inconsistent. If we really need to consider all these three values, we need to do it consistently, in a single place, and make sure all those values are considered every time any of them are updated.

Reply:
Ya it is kinda hard to follow. I'm not 100% sure the best way to go about this. If you want I can port this PR to an issue and link this as a potential solution so that the best way to fix this can be discussed outside of the context of a specific implementation.

Reply:
I'm not 100% sure, but I think there is only the `invalid` property and the valid status of the native element. Material is using the native validity reported as well if I remember correctly. The `invalid` property should probably always have precedence over anything reported by the element

Reply:
> If you want I can port this PR to an issue and link this as a potential solution

Yeah, that would be great. It's generally a good idea to create a bug report first, even if you already have a PR ready to go (not that I always do it myself 😖).

Reply:
issue: https://github.com/Lundalogik/lime-elements/issues/791